generators/base/templates/App/Components/Authentication/index.tsx

```diff
@@ -0,0 +1,86 @@
+import React, { createContext, useContext, useEffect, useReducer } from "react";
+import Keychain from "<%= name %>/App/Helpers/Keychain";
+import reducer, { AuthenticationState } from "./reducer";
+
+const KEYCHAIN_KEY = "@authenticationToken";
+
+const AuthenticationContext = createContext<AuthenticationContextState | null>(
+  null,
+);
+
+export function AuthenticationProvider({
+  children,
+}: AuthenticationProviderProps) {
+  const [state, dispatch] = useReducer(reducer, {
+    isLoading: true,
+    isLoggedIn: false,
+  });
+
+  const login: LoginFunction = async (fields) => {
+    // @TODO replace this hard coded token from an API
+    const token: AuthenticationToken = {
+      access_token: "romulus",
```

Comment:
Does this need to be `"<%= name %>"`?

Reply:
Ah no, it’s intentionally there as a placeholder so the login can work. It’s then intended to be replaced by an oauth token or something.